### PR TITLE
CPLAT-12081 Readme Improvements

### PR DESCRIPTION
**Pull request by aaronlademann-wf**
## Motivation
As I was doing a full in-depth read through our README this morning following the awesome 3.5.0 release, I noticed a few things that were wrong, or that just needed a little love/updating.

## Changes
Misc. updates and improvements to the README. I broke it all up into fairly granular commits if you want to digest the changes that way.

#### Release Notes
Improve documentation

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: @greglittlefield-wf @sydneyjodon-wk @joebingham-wk 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: README.md [hunk 1]
```diff
@@ -503,10 +537,16 @@ ReactElement renderResizeHandle() {
 * OverReact DOM components return a new `DomProps` builder, which can be used
 to render them via our [fluent interface](#fluent-style-component-consumption)
 as shown in the examples above.
-* `DomProps` has statically-typed getters and setters for all "ubiquitous" HTML attribute props.
-  * The `domProps()` function is also available to create a new typed Map or a typed view into an
-  existing Map. Useful for manipulating DOM props and adding DOM props to components that don’t
-  forward them directly.
+  * `DomProps` has statically-typed getters and setters for all "ubiquitous" HTML attribute props.
```
Comment (greglittlefield-wf) [inline README.md:540]: While we're in here
```suggestion
  * `DomProps` has statically-typed getters and setters for all HTML attribute props.
```

File: README.md [hunk 2]
```diff
@@ -503,10 +537,16 @@ ReactElement renderResizeHandle() {
...
+  * The `domProps()` function is also available to create a new typed Map or a typed view into an existing Map. Useful for manipulating DOM props and adding DOM props to components that don’t forward them directly, or to access a dom prop from a plain map in a lifecycle method as shown below.
```
Comment (greglittlefield-wf) [inline README.md:541]: #nit
```suggestion
  * The `domProps()` function is also available to create a new typed Map or a typed view into an existing Map. Useful for manipulating DOM props and adding DOM props to components that don’t forward them directly, or to access a DOM prop from a plain map in a lifecycle method as shown below.
```

File: README.md [hunk 3]
```diff
@@ -730,7 +768,6 @@ that you get for free from OverReact, you're ready to start building your own cu
 * #### Stateful Component Boilerplate
 
     ```dart
-    import 'dart:html';
     import 'package:over_react/over_react.dart';
     part 'foo_component.over_react.g.dart';
```
Comment (greglittlefield-wf) [inline README.md:773]: Ooh if you're in here editing this, we probably want to add the ignore comments to factories
```diff
UiFactory<BarProps> Bar = _$Bar; // ignore: undefined_identifier
```

File: doc/props_mixin_component_composition.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
+# Composing Props Mixins Into A Single Component Props API
+
+In our core `UiProps` documentation, the pattern of [composing multiple props mixins into a single component props API](../README.md#with-other-mixins) is introduced.
+
+This example builds on that, showing a lightweight example a common use-case for such composition. 
+
+We'll show two components 
+
+1. A `Bar` component that has its own props API - and default rendering behavior when rendered standalone.
+2. A `FooBar` component that has its own props API in addition to the `Bar` props API that it will use to allow consumers to compose `Bar` with `FooBar` by forwarding props from `BarPropsMixin` to the `Bar` component it renders.
```
Comment (greglittlefield-wf) [inline doc/props_mixin_component_composition.md:10]: Run-on sentence
```suggestion
2. A `FooBar` component that has its own props API, in addition to the `Bar` props API. This allows consumers to set props declared in `BarPropsMixin`, which will be forwarded to to the `Bar` component it renders.
```

File: doc/props_mixin_component_composition.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+### Bar Component
+```dart
+import 'package:over_react/over_react.dart';
+
+UiFactory<BarPropsMixin> Bar = _$Bar;
```
Comment (greglittlefield-wf) [inline doc/props_mixin_component_composition.md:16]: Missing part and ignore comment
```suggestion
import 'package:over_react/over_react.dart';

part 'bar.over_react.g.dart';

UiFactory<BarPropsMixin> Bar = _$Bar; // ignore: undefined_identifier
```

File: doc/props_mixin_component_composition.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+```
+
+### Composing Bar using the FooBar component
+
+To compose the `Bar` component using `FooBar`, we'll expose the prop API for both the `BarPropsMixin` and the `FooPropsMixin` like so:
+```dart
+import 'package:over_react/over_react.dart';
+import 'bar.dart';
+
+UiFactory<FooBarProps> FooBar = _$FooBar;
```
Comment (greglittlefield-wf) [inline doc/props_mixin_component_composition.md:55]: Missing part and ignore comment
```suggestion
part 'foo_bar.over_react.g.dart';

UiFactory<FooBarProps> FooBar = _$FooBar; // ignore: undefined_identifier
```

**Comment by aaronlademann-wf** [comment]
Sorry I lost track of this @greglittlefield-wf - all feedback has been addressed.

**Review by greglittlefield-wf — APPROVED** [review]
+10

@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**